src/hooks/detectUserDevice.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react'
+
+/** Custom hook detectUserDevice.ts to detect user device. */
+const DetectUserDevice = () => {
```

Comment:
For future reference, hooks should always start with `use`, e.g. `useDetectUserDevice`.